src/libraries/System.Text.Json/tests/Common/CollectionTests/CollectionTests.Dictionary.NonStringKey.cs

```diff
@@ -5,526 +5,296 @@
-            protected override int Value => 1;
-        }
+            const string expected = @"{""1"":1,""08314fa2-b1fe-4792-bcd1-6e62338ac7f3"":2,""KeyString"":3,""Foo"":4,""Foo, Bar"":5}";
 
-        public class DictionarySingleKey : DictionaryKeyTestsBase<float, int>
-        {
-            protected override string _expectedJson => $@"{{""{JsonSerializer.Serialize(float.MaxValue)}"":1}}";
-            protected override float Key => float.MaxValue;
-            protected override int Value => 1;
-        }
+            string json = await Serializer.SerializeWrapper(dictionary);
+            Assert.Equal(expected, json);
+            // object type is not supported on deserialization.
+            await Assert.ThrowsAsync<NotSupportedException>(() => Serializer.DeserializeWrapper<Dictionary<object, object>>(json));
 
-        public class DictionaryStringKey : DictionaryKeyTestsBase<string, int>
-        {
-            protected override string Key => "KeyString";
-            protected override int Value => 1;
+            var @object = new ClassWithDictionary { Dictionary = dictionary };
+            json = await Serializer.SerializeWrapper(@object);
+            Assert.Equal($@"{{""Dictionary"":{expected}}}", json);
+            await Assert.ThrowsAsync<NotSupportedException>(() => Serializer.DeserializeWrapper<ClassWithDictionary>(json));
         }
 
-        public class DictionaryUInt16Key : DictionaryKeyTestsBase<ushort, int>
+        [Fact]
+        public async Task TestNonGenericDictionaryKeyObject()
         {
-            protected override ushort Key => ushort.MaxValue;
-            protected override int Value => 1;
-        }
+            IDictionary dictionary = new OrderedDictionary();
+            // Add multiple supported types.
+            dictionary.Add(1, 1);
+            dictionary.Add(new Guid("08314FA2-B1FE-4792-BCD1-6E62338AC7F3"), 2);
+            dictionary.Add("KeyString", 3);
+            dictionary.Add(MyEnum.Foo, 4);
+            dictionary.Add(MyEnumFlags.Foo | MyEnumFlags.Bar, 5);
 
-        public class DictionaryUInt32Key : DictionaryKeyTestsBase<uint, int>
-        {
-            protected override uint Key => uint.MaxValue;
-            protected override int Value => 1;
-        }
+            const string expected = @"{""1"":1,""08314fa2-b1fe-4792-bcd1-6e62338ac7f3"":2,""KeyString"":3,""Foo"":4,""Foo, Bar"":5}";
+            string json = await Serializer.SerializeWrapper(dictionary);
+            Assert.Equal(expected, json);
 
-        public class DictionaryUInt64Key : DictionaryKeyTestsBase<ulong, int>
-        {
-            protected override ulong Key => ulong.MaxValue;
-            protected override int Value => 1;
-        }
+            dictionary = await Serializer.DeserializeWrapper<IDictionary>(json);
+            Assert.IsType<Dictionary<string, object>>(dictionary);
 
-        public abstract class DictionaryUnsupportedKeyTestsBase<TKey, TValue>
```

Comment:
I see - removing this class and replacing with theories.

Reply:
There's a lot of changes in this file - I did a few spot checks to ensure the tests are still there, but didn't verify everything.